### PR TITLE
Add `Function::{ScriptId, GetScriptOrigin}, ScriptOrigin::Get{ScriptId, ResourceName, SourceMapUrl}` bindings

### PR DESCRIPTION
**Pull request by rakeeb-hossain**
I added the bindings in the title to use in an application! Let me know if I'm missing anything.

### CODE REVIEW

File: src/function.rs [hunk 1]
```diff
@@ -903,6 +907,24 @@ impl Function {
...
+  pub fn get_script_origin(&self) -> &ScriptOrigin {
+    unsafe {
+      let ptr = v8__Function__GetScriptOrigin(self);
+      &*ptr
+    }
+  }
+
+  /// Returns scriptId.
+  #[inline(always)]
+  pub fn get_script_id(&self) -> Option<u32> {
```
Comment (mmastrac) [inline src/function.rs:920]: Should this be i32?

Comment (rakeeb-hossain) [reply src/function.rs:920]: Was unsure about this. V8 docs are sparse on when a script ID is < 0, but the only cases I could find in the source were 1) uninitialized ScriptOrigin classes (value is -1) and 2) kTemporaryScriptId = -2. Due to uncertainty on when these cases actually occur, I just treated these as None values.

I can change this to i32 instead if you think that would make more sense.

Comment (rakeeb-hossain) [reply src/function.rs:920]: @mmastrac Would you suggest i32 instead?

Comment (bartlomieju) [reply src/function.rs:920]: Yes, we should use `i32` here. Also please remove `get_` prefix - in `v8/include/v8-function.h` this function is named `ScriptId()`

File: src/script.rs [hunk 2]
```diff
@@ -125,4 +132,28 @@ impl<'s> ScriptOrigin<'s> {
...
+  #[inline(always)]
+  pub fn get_resource_name(&self) -> Option<Local<'s, Value>> {
+    unsafe {
+      let ptr = v8__ScriptOrigin__GetResourceName(self);
+      Local::from_raw(ptr)
+    }
+  }
+
+  #[inline(always)]
+  pub fn get_source_map_url(&self) -> Option<Local<'s, Value>> {
```
Comment (bartlomieju) [inline src/script.rs:153]: Please remove `get_` prefixes from these functions to match V8 API

File: src/script.rs [hunk 3]
```diff
@@ -125,4 +132,28 @@ impl<'s> ScriptOrigin<'s> {
       buf.assume_init()
     }
   }
+
+  #[inline(always)]
+  pub fn get_script_id(&self) -> Option<u32> {
```
Comment (bartlomieju) [inline src/script.rs:137]: Use `i32` here

**Review by bartlomieju — COMMENTED** [review]
Almost there, just a few nitpicks to match V8 API

File: src/function.rs (hunk 1 above)
Comment (rakeeb-hossain) [reply src/function.rs]: Done @bartlomieju @mmastrac

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you!

**Pull request merged**